Content.Shared/_Impstation/Fishing/SharedFishingSystem.cs

```diff
@@ -47,7 +53,105 @@ public override void Initialize()
         SubscribeLocalEvent<FishingProjectileComponent, JointRemovedEvent>(OnGrappleJointRemoved);
         SubscribeLocalEvent<FishingProjectileComponent, RemoveEmbedEvent>(OnRemoveEmbed);
     }
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        #region chumming
+        var queryBait = EntityQueryEnumerator<FishingBaitComponent>();
+
+        //For each bait
+        while (queryBait.MoveNext(out var uid, out var baitComp))
+        {
+            //Get the bait's transform component
+            if (TryComp<TransformComponent>(uid, out var transformComp))
+            {
+                //Check if it's spaced
+                if (transformComp.GridUid == null)
+                {
+                    //Increase the fishing timer
+                    baitComp.BaitTimer += frameTime;
+
+                    //At the given time, reset the timer and try a catch
+                    if (baitComp.BaitTimer > baitComp.CatchAttemptTime)
+                    {
+                        baitComp.BaitTimer = 0;
+
+                        //Chance a catch
+                        if (_robustRandom.Prob(Math.Min(baitComp.CatchChance, 1.0f)))
```

Comment:
this could be refactored per the comment in `FishingBaitComponent` to use `WeightedRandom` instead of its own code